File: C/203_removeElements.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeElements(struct ListNode* head, int val) {
    struct ListNode *temp = NULL;
    struct ListNode *cur = NULL;
    
    while(head && head->val == val) {
        temp = head;
        head = head->next;
        free(temp);
    }
    
    cur = head;
    while(cur) {
        if (cur->next && cur->next->val == val) {
            temp = cur->next;
            cur->next = temp->next;
            free(temp);
        } else 
            cur = cur->next;
    }
    
    return head;

File: python/203_removeElements.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        while head and head.val == val:
            head = head.next
        
        cur = head
        while cur:
            if cur.next and cur.next.val == val:
                cur.next = cur.next.next
            else:
                cur = cur.next
        
        return head
